processor/tailsampling/processor.go

```diff
@@ -69,40 +72,41 @@ const (
 	sourceFormat = "tail-sampling"
 )
 
-var _ consumer.TraceConsumer = (*tailSamplingSpanProcessor)(nil)
+var _ processor.TraceProcessor = (*tailSamplingSpanProcessor)(nil)
 
-// NewTailSamplingSpanProcessor creates a TailSamplingSpanProcessor with the given policies.
-// It will keep maxNumTraces on memory and will attempt to wait until decisionWait before evaluating if
-// a trace should be sampled or not. Providing expectedNewTracesPerSec helps with allocating data structures
-// with closer to actual usage size.
-func NewTailSamplingSpanProcessor(
-	policies []*Policy,
-	maxNumTraces, expectedNewTracesPerSec uint64,
-	decisionWait time.Duration,
-	logger *zap.Logger) (consumer.TraceConsumer, error) {
+// NewTraceProcessor returns a processor.TraceProcessor that will perform tail sampling according to the given
+// configuration.
+func NewTraceProcessor(logger *zap.Logger, nextConsumer consumer.TraceConsumer, cfg builder.TailBasedCfg) (processor.TraceProcessor, error) {
+	if nextConsumer == nil {
+		return nil, errors.New("nextConsumer is nil")
+	}
 
-	numDecisionBatches := uint64(decisionWait.Seconds())
-	inBatcher, err := idbatcher.New(numDecisionBatches, expectedNewTracesPerSec, uint64(2*runtime.NumCPU()))
+	numDecisionBatches := uint64(cfg.DecisionWait.Seconds())
+	inBatcher, err := idbatcher.New(numDecisionBatches, cfg.ExpectedNewTracesPerSec, uint64(2*runtime.NumCPU()))
 	if err != nil {
 		return nil, err
 	}
+
 	tsp := &tailSamplingSpanProcessor{
 		ctx:             context.Background(),
-		maxNumTraces:    maxNumTraces,
-		policies:        policies,
+		nextConsumer:    nextConsumer,
+		maxNumTraces:    cfg.NumTraces,
 		logger:          logger,
 		decisionBatcher: inBatcher,
+		// policies:        policies,
 	}
 
-	for _, policy := range policies {
-		policyCtx, err := tag.New(tsp.ctx, tag.Upsert(tagPolicyKey, policy.Name), tag.Upsert(observability.TagKeyReceiver, sourceFormat))
-		if err != nil {
-			return nil, err
-		}
-		policy.ctx = policyCtx
-	}
+	// TODO(#146): add policies to TailBasedCfg
```

Comment:
Policies have an interface and we will need to add the config factories for each one (as a temporary alternative we can use the old code to read them but I want to double-check that before going in that direction).

Reply:
Got it, either way I would prefer to add it in a follow-up PR.